rocketpy/Function.py

```diff
@@ -2227,11 +2231,31 @@ def integral(self, a, b, numerical=False):
                     # self.__extrapolation__ = 'zero'
                     pass
         elif self.__interpolation__ == "linear" and numerical is False:
-            return np.trapz(self.source[:, 1], x=self.source[:, 0])
+            # Integrate from a to b using np.trapz
+            xData = self.source[:, 0]
+            yData = self.source[:, 1]
+            # Get data in interval
+            xIntegrationData = xData[(xData >= a) & (xData <= b)]
+            yIntegrationData = yData[(xData >= a) & (xData <= b)]
+            # Add integration limits to data
+            if self.__extrapolation__ == "zero":
```

Comment:
The idea of adding the integration limits to the data was interesting for consistency, however it introduces new spurious results when integrating with `np.trapz`. See below:

```
>>> from rocketpy import Function
>>> f = Function([(0,0), (1,2), (10,20)], interpolation="linear", extrapolation="zero")
>>> f.integral(0,11) # "exact" integral
110.0  # should be 100.0
>>> f.integral(0,11,True)
100.00083463213139
```

The problem is that the integral limit in the zone of zero extrapolation is causing the trapezoid rule to interpret one more region as a trapezoid whose lesser base is 0 (i.e. a triangle).

I do not have a nice fix for the problem, however, if the integration limit is not added to the dataset, the function gives the correct value.

Reply:
A typo in line 2245, where `yData` should be `xData` caused the problem. I have fixed it in the latest commit. Therefore, I am requesting your review once again @phmbressan